### PR TITLE
Display visible indicate the user could not be found

### PR DESCRIPTION
**Pull request by kelfink**
#### Which User story does this relate to (link)?
-
https://osi-cwds.atlassian.net/browse/COG-743
#### Brief feature description
-  Failing to load a user record (valid problem or bug) should not show an empty page. Some indicator need to show the user that the user wasn't found.

### CODE REVIEW

File: app/javascript/views/userDetail/UserDetail.jsx [hunk 1]
```diff
@@ -136,41 +139,52 @@ export default class UserDetail extends Component {
   };
 
   renderCards = permissionsList => {
+    console.log('state rendering cards', this.state);
```
Comment (Nasir2712) [inline app/javascript/views/userDetail/UserDetail.jsx]: remove console.log

File: app/javascript/containers/detailsContainer.js [hunk 2]
```diff
@@ -24,6 +25,7 @@ import { officesList } from '../selectors/userListSelector';
 
 function mapStateToProps(state) {
   return {
+    XHRStatus: fetchingStatus(state),
```
Comment (Nasir2712) [inline app/javascript/containers/detailsContainer.js]: we could just use fetching in the props as boolean.

File: app/javascript/selectors/detailSelector.js [hunk 3]
```diff
@@ -22,3 +22,6 @@ export const rolesList = state => safeGet(state, 'fetchRoles.roles', []);
 
 export const checkEditDisable = state =>
   !safeGet(state, 'fetchDetails.details.records.editable');
+
+export const fetchingStatus = state =>
```
Comment (Nasir2712) [inline app/javascript/selectors/detailSelector.js]: we probably won't need this.

File: app/javascript/views/userDetail/UserDetail.jsx [hunk 4]
```diff
@@ -201,7 +215,13 @@ export default class UserDetail extends Component {
             {this.state.disableErrorAlert &&
               userDetailError && <ErrorMessage error={userDetailError} />}
           </div>
-          {this.renderCards(permissionsList)}
+          {this.state.XHRStatus === 'ready' ? (
```
Comment (Nasir2712) [inline app/javascript/views/userDetail/UserDetail.jsx]: this can be wrapped inside a method to avoid conditional rendering.

Comment (kelfink) [reply app/javascript/views/userDetail/UserDetail.jsx]: Yeah I can wrap it inside renderCards itself

File: app/javascript/selectors/detailSelector.js (hunk 3 above)
Comment (kelfink) [reply app/javascript/selectors/detailSelector.js]: Fetching is whether it's in process.  XHRStatus is whether it was successful

Comment (Nasir2712) [reply app/javascript/selectors/detailSelector.js]: Ya, wouldn't it make sense to show the user the page is loading and then display the results if they exist or not?

Comment (kelfink) [reply app/javascript/selectors/detailSelector.js]: ^ that's what this fix does..... "Loading..." shows, then the record.  Or "User could not be found" if we're not in the process of fetching and there's no detail.

Comment (Nasir2712) [reply app/javascript/selectors/detailSelector.js]: fetching means it's in the process, so shouldn't that indicate loading card?

Comment (kelfink) [reply app/javascript/selectors/detailSelector.js]: While the saga is running, it shows "Loading..."

File: app/javascript/containers/detailsContainer.js (hunk 2 above)
Comment (kelfink) [reply app/javascript/containers/detailsContainer.js]: the fetching boolean just tells us if it is actively fetching, not whether it was successful when done

File: app/javascript/selectors/detailSelector.js (hunk 3 above)
Comment (Nasir2712) [reply app/javascript/selectors/detailSelector.js]: had an offline conversation, explanation understood why another state is needd. Thanks for the clarification.

File: app/javascript/views/userDetail/UserDetail.jsx [hunk 5]
```diff
@@ -138,41 +141,56 @@ export default class UserDetail extends Component {
   };
 
   renderCards = permissionsList => {
-    return (
+    console.log(this.state);
```
Comment (Nasir2712) [inline app/javascript/views/userDetail/UserDetail.jsx]: this needs to be removed

File: app/javascript/views/userDetail/UserDetail.jsx [hunk 6]
```diff
@@ -138,41 +141,56 @@ export default class UserDetail extends Component {
...
-                disableEditBtn={this.props.disableEditBtn}
-                officesList={this.props.officesList}
-                rolesList={this.props.rolesList}
-              />
-            )}
-          </div>
-        ) : (
-          ''
-        )}
+        <div>
```
[the remark this answers is not in the document]
Comment (kelfink) [reply app/javascript/views/userDetail/UserDetail.jsx]: Good question.  Don't know.  Removed.  Looks like it was invisible anyway.

File: app/javascript/views/userDetail/UserDetail.jsx (hunk 5 above)
Comment (sunilb0575) [reply app/javascript/views/userDetail/UserDetail.jsx]: Also please remove this console.log.

**Pull request merged**